types/lodash/lodash-tests.ts

```diff
@@ -3656,25 +3656,56 @@ fp.now(); // $ExpectType number
 
 // _.throttle
 {
-    const options: _.ThrottleSettings = {
+    const optionsLeading: _.ThrottleSettingsLeading = {
         leading: true,
         trailing: false,
+    }
+    const badOptionsLeading: _.ThrottleSettingsLeading = {
+        // @ts-expect-error lodash will treat an explicit undefined the same as `leading: false`
+        leading: undefined,
+    };
+    const optionsNoLeading: _.ThrottleSettings = {
+        leading: false,
+    };
+
+    const optionsAmbiguous = {
+        leading: true,
     };
+    const boolean: boolean = false;
```

Comment:
nit: you might need `false as boolean`, otherwise TS will automatically narrow the type to `false` which is not what we want.